src/Features/Core/Portable/EditAndContinue/EditAndContinueService.cs

```diff
@@ -118,5 +165,81 @@ public bool IsProjectReadOnly(ProjectId id, out SessionReadOnlyReason sessionRea
             projectReason = ProjectReadOnlyReason.MetadataNotAvailable;
             return true;
         }
+
+        public async Task<LinePositionSpan?> GetCurrentActiveStatementPositionAsync(ActiveInstructionId instructionId, CancellationToken cancellationToken)
+        {
+            try
+            {
+                // It is allowed to call this method before entering or after exiting break mode. In fact, the VS debugger does so. 
+                // We return null since there the concept of active statement only makes sense during break mode.
+                if (_editSession == null)
+                {
+                    return null;
+                }
+
+                Debug.Assert(_debuggingSession != null);
+
+                var baseActiveStatements = await _editSession.BaseActiveStatements.GetValueAsync(cancellationToken).ConfigureAwait(false);
+                if (!baseActiveStatements.InstructionMap.TryGetValue(instructionId, out var baseActiveStatement))
+                {
+                    return null;
+                }
+
+                Document primaryDocument = _debuggingSession.InitialSolution.Workspace.CurrentSolution.GetDocument(baseActiveStatement.PrimaryDocumentId);
+
+                // TODO:is this needed?
+                // Try to get spans from the tracking service first.
+                // We might get an imprecise result if the document analysis hasn't been finished yet and 
+                // the active statement has structurally changed, but that's ok. The user won't see an updated tag
+                // for the statement until the analysis finishes anyways.
+                //
+                // SourceText text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                //if (_trackingService.TryGetSpan(baseActiveStatement.Id, text, out var trackedSpan) && trackedSpan.Length > 0)
+                //{
+                //    lineSpan = text.Lines.GetLinePositionSpan(trackedSpan);
+                //}
```

Comment:
obsolete?

Reply:
Deleted.

---
In reply to: [162477526](https://github.com/dotnet/roslyn/pull/24316#discussion_r162477526) [](ancestors = 162477526)